e2e/ethereum/ethereum.go

```diff
@@ -140,21 +138,21 @@ func CreateErc20Deposit(t *testing.T, client *ethclient.Client, opts *bind.Trans
 	}
 }
 
-func CreateErc721Deposit(t *testing.T, client *ethclient.Client, opts *bind.TransactOpts, destId msg.ChainId, recipient []byte, tokenId *big.Int, contracts *utils.DeployedContracts, rId msg.ResourceId) {
+func CreateErc721Deposit(t *testing.T, client *utils.Client, destId msg.ChainId, recipient []byte, tokenId *big.Int, contracts *utils.DeployedContracts, rId msg.ResourceId) {
 	data := utils.ConstructErc721DepositData(rId, tokenId, recipient)
 
-	bridgeInstance, err := bridge.NewBridge(contracts.BridgeAddress, client)
+	bridgeInstance, err := bridge.NewBridge(contracts.BridgeAddress, client.Client)
```

Comment:
I only pass in client.Client (type ethClient) because we don't need everything from client

Reply:
Not much choice here, these methods are part a library that is auto-generated from out solidity code. These constructors are standard, the method calls like `Deposit` below have a standard format (ie. they require `opts`) and then the parameters for the solidity function are included as well.